src/protocolgame.cpp

```diff
@@ -3570,8 +3570,8 @@ void ProtocolGame::AddShopItem(NetworkMessage& msg, const ShopInfo& item)
 
 	msg.addString(item.realName);
 	msg.add<uint32_t>(it.weight);
-	msg.add<uint32_t>(item.buyPrice == std::numeric_limits<uint32_t>::max() ? 0 : item.buyPrice);
-	msg.add<uint32_t>(item.sellPrice == std::numeric_limits<uint32_t>::max() ? 0 : item.sellPrice);
+	msg.add<uint32_t>(item.buyPrice == std::numeric_limits<int64_t>::max() ? 0 : item.buyPrice);
+	msg.add<uint32_t>(item.sellPrice == std::numeric_limits<int64_t>::max() ? 0 : item.sellPrice);
```

Comment:
`-1` in both cases

Reply:
isn't that going to affect the check here?
https://github.com/otland/forgottenserver/blob/33bca9feb5d3e0bf03f450f035cbcc7590b054ae/src/protocolgame.cpp#L1810

Reply:
I meant, use -1 instead of `std::numeric_limits<int64_t>::max()`. It would work previously because `static_cast<uint32_t>(-1) == std::numeric_limits<uint32_t>::max()` but it is not true with `int64_t`

Reply:
honestly I would drop these checks entirely, why are we checking that?
use `std::max<uint32_t>(item.sellPrice, 0);`?

Reply:
Even better